cli/src/commands/node/mod.rs

```diff
@@ -188,10 +185,20 @@ impl Node {
         let rng_seed = rng.next_u64();
         let srs: Arc<_> = get_srs();
 
-        let transition_frontier = match conf {
-            Some(c) => TransitionFrontierConfig::new(Arc::new(GenesisConfig::DaemonJson(c))),
-            None => TransitionFrontierConfig::new(node::config::BERKELEY_CONFIG.clone()),
+        let genesis_config = match conf {
+            Some(c) => Arc::new(GenesisConfig::DaemonJson(c)),
+            None => node::config::BERKELEY_CONFIG.clone(),
         };
+        let transition_frontier = TransitionFrontierConfig::new(genesis_config.clone());
+        let protocol_constants = genesis_config.protocol_constants()?;
+        let chain_id = ChainId::compute(
+            constants::CONSTRAINT_SYSTEM_DIGESTS.as_slice(),
+            &constants::GENESIS_STATE_HASH,
```

Comment:
Just to make sure... Is it constant in OCaml?

Reply:
It's a part of network configuration. As far as I could see, it's contrlled by a dune profile in OCaml.